lib/sles4sap_publiccloud.pm

```diff
@@ -432,18 +435,51 @@ sub wait_for_pacemaker {
 
      "no" - do not set and turn off SBD delay time
      "yes" - sets default SBD value which is calculated from a formula
-     "<number of seconds>" - sets sepcific delay in seconds
+     "<number of seconds>" - sets specific delay in seconds
+
+     Returns integer representing wait time.
 
 =cut
 
 sub setup_sbd_delay() {
-    my ($self, $set_delay) = @_;
-    my $delay = $set_delay || "no";
-    record_info("SBD delay", "Setting SBD delay to: $delay");
+    my ($self) = @_;
+    my $delay = get_var('HA_SBD_START_DELAY');
+    $delay =~ s/[^ye]s//g;
+    if (get_var('HA_SBD_START_DELAY') eq '') {
+        record_info('SBD delay', 'Skipping, parameter without value');
+        return;
+    }
+
+    croak("<\$set_delay> value must be either 'yes', no or an integer. Got value: $delay")
+      unless looks_like_number($delay) or grep /^$delay$/, qw(yes no);
+
     $self->cloud_file_content_replace('/etc/sysconfig/sbd', '^SBD_DELAY_START=.*', "SBD_DELAY_START=$delay");
+    record_info('SBD delay', "SBD delay set to: $delay");
     return 1;
 }
 
+=head2 sbd_delay_formula
+    sbd_delay_formula();
+
+
+=cut
+
+sub sbd_delay_formula() {
+    my ($self) = @_;
+    my %params = (
+        'corosync_token' => $self->run_cmd(cmd => "corosync-cmapctl | awk -F \" = \" '/runtime.config.totem.token\\s/ {print \$2 /1000}'"),
```

Comment:
Yep. as you mentioned in other comment I added those command as global ones in hacluster.pm and exported them:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/537dcd65f0173dd087114d3b38dc88adbedcaeee/lib/hacluster.pm#L106

I am running two VRs now, one for PC and one for on prem. 
https://openqa.suse.de/tests/11237258#live
https://openqaworker15.qa.suse.cz/tests/185484#live

Reply:
Both tests finished. PC failed but because of the timeout.

Reply:
> Both tests finished. PC failed but because of the timeout.

Seems on prem test failed with a syntax error: https://openqa.suse.de/tests/11237258#step/check_after_reboot/25

Haven't looked at the code yet, so don't know if you already fixed it. Just leaving the note here just in case.

Reply:
Ah sorry ... forgotten to update the VR link https://openqa.suse.de/tests/11242888
It's fixed now.